packages/widgets/user-profile-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initUserCustomAttributesMixin.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  FlowDriver,
+  ModalDriver,
+  UserAttributeDriver,
+} from '@descope/sdk-component-drivers';
+import {
+  compose,
+  createSingletonMixin,
+  withMemCache,
+} from '@descope/sdk-helpers';
+import { loggerMixin, modalMixin } from '@descope/sdk-mixins';
+import { getUserCustomAttrs } from '../../../state/selectors';
+import { createFlowTemplate } from '../../helpers';
+import { stateManagementMixin } from '../../stateManagementMixin';
+import { initWidgetRootMixin } from './initWidgetRootMixin';
+
+export const initUserCustomAttributesMixin = createSingletonMixin(
+  <T extends CustomElementConstructor>(superclass: T) =>
+    class UserCustomAttributesMixinClass extends compose(
+      stateManagementMixin,
+      loggerMixin,
+      initWidgetRootMixin,
+      modalMixin,
+    )(superclass) {
+      customValueUserAttr: UserAttributeDriver;
+
+      // flow Id is key in all maps
+      #editModals: Record<string, ModalDriver> = {};
+
+      #editFlows: Record<string, FlowDriver> = {};
+
+      #deleteModals: Record<string, ModalDriver> = {};
+
+      #deleteFlows: Record<string, FlowDriver> = {};
+
+      #initEditModalContent(flowId: string) {
+        this.#editModals[flowId]?.setContent(
+          createFlowTemplate({
+            projectId: this.projectId,
+            flowId,
+            baseUrl: this.baseUrl,
+            baseStaticUrl: this.baseStaticUrl,
+          }),
+        );
+        this.#editFlows[flowId]?.onSuccess(() => {
+          this.#editModals[flowId]?.close();
+          this.actions.getMe();
+        });
+      }
+
+      // have 2 init functions for edit and delete modals in order to keep the same standards as the email/phone/name mixin
+      #initDeleteModalContent(flowId: string) {
+        this.#deleteModals[flowId]?.setContent(
+          createFlowTemplate({
+            projectId: this.projectId,
+            flowId,
+            baseUrl: this.baseUrl,
+            baseStaticUrl: this.baseStaticUrl,
+          }),
+        );
+        this.#deleteFlows[flowId]?.onSuccess(() => {
+          this.#deleteModals[flowId]?.close();
+          this.actions.getMe();
+        });
+      }
+
+      #updateCustomValueUserAttrs = withMemCache(
```

Comment:
This function is a bit big and contains some repetitive code
Do you think we could split it into smaller functions?

Reply:
It was really about keeping the same standards as the other mixin - to begin with a preferred extracting code to functions and reuse.
Let me see what I can do here